xdsl/printer.py

```diff
@@ -702,3 +705,20 @@ def print_op(self, op: Operation) -> None:
             op.print(self)
         else:
             self.print_op_with_default_format(op)
+
+    def print_optional_attr(
```

Comment:
can you make separate pull requests for the change here and the one above? Depending on the one you keep in this PR, can you please rename the PR into something more descriptive?

Reply:
Once I finish updating mlir I will make the new pr

Reply:
Can you mark it as draft plz?